### PR TITLE
[AST Builder] Implement to_date and let format function to format date data type

### LINKED ISSUE #854: [AST Builder] Implement `to_date` and let format function to format date data type
As the new function `TO_DATE` and the `FORMAT` supports the new data type `date` I think we need to implement these functions into `ast_builder`. The following functions should pass 

```rust
    #[test]
    fn function_format() {
        let actual = format(time("23:56:04"), text("%H:%M:%S"));
        let expected = "FORMAT(TIME '23:56:04', '%H:%M:%S')";
        test_expr(actual, expected);

        let actual = time("23:56:04").format(text("%H:%M:%S"));
        let expected = "FORMAT(TIME '23:56:04', '%H:%M:%S')";
        test_expr(actual, expected);
    }

```

```rust
    #[test]
    fn function_to_time() {
        let actual = to_time(text("23:56:04"), text("%H:%M:%S"));
        let expected = "TO_TIME('23:56:04','%H:%M:%S')";
        test_expr(actual, expected);

        let actual = text("23:56:04").to_time(text("%H:%M:%S"));
        let expected = "TO_TIME('23:56:04','%H:%M:%S')";
        test_expr(actual, expected);
    }
````

### PR DESCRIPTION
**Pull request by ChobobDev**
resolves #854

### CODE REVIEW

File: core/src/ast_builder/expr/mod.rs [hunk 1]
```diff
@@ -294,3 +294,10 @@ pub fn timestamp(timestamp: &str) -> ExprNode {
         value: (timestamp.to_string()),
     })
 }
+
+pub fn time(time: &str) -> ExprNode {
+    ExprNode::Expr(Expr::TypedString {
+        data_type: (DataType::Time),
+        value: (time.to_string()),
```
Comment (ever0de) [inline core/src/ast_builder/expr/mod.rs:301]: Remove '()' brackets

Comment (ChobobDev) [reply core/src/ast_builder/expr/mod.rs:301]: I will apply this changes ASAP ,

thank you

**Review by ever0de — APPROVED** [review]
Looks good

**Review by panarch — APPROVED** [review]
Thanks a lot 👍
Looks all good

**Pull request merged**